FprimeZephyrReference/Components/AuthenticationRouter/AuthenticationRouter.cpp

```diff
@@ -153,6 +168,69 @@ void AuthenticationRouter ::cmdResponseIn_handler(FwIndexType portNum,
     // Nothing to do
 }
 
+void AuthenticationRouter ::run_handler(FwIndexType portNum, U32 context) {
+    Fw::Time command_loss_start = this->update_command_loss_start();
+
+    Fw::ParamValid is_valid;
+    Fw::TimeIntervalValue command_loss_period = this->paramGet_COMM_LOSS_TIME(is_valid);
+    FW_ASSERT(is_valid == Fw::ParamValid::VALID || is_valid == Fw::ParamValid::DEFAULT);
+    Fw::Time command_loss_interval(command_loss_start.getTimeBase(), command_loss_period.get_seconds(),
+                                   command_loss_period.get_useconds());
+    Fw::Time command_loss_end = Fw::Time::add(command_loss_start, command_loss_interval);
+
+    Fw::Time current_time =
+        (command_loss_end.getTimeBase() == TimeBase::TB_PROC_TIME) ? this->get_uptime() : this->getTime();
```

Comment:
What is the reason for doing this? If the time base is `TimeBase::TB_PROC_TIME`, the FPrime getTime() fn will return uptime.